File: CareConnect/CareConnect.Models/Responses/Participant.cs
using System; 

namespace CareConnect.Models.Responses
{
    public class Participant
    {
        public int ParticipantId { get; set; }

        public DateTime RegistrationDate { get; set; }

        public DateTime ModifiedDate { get; set; }

        public int AttendanceStatusId { get; set; }

        public virtual AttendanceStatus AttendanceStatus { get; set; } = null!;

        public virtual User User { get; set; } = null!;

        public virtual Workshop Workshop { get; set; } = null!;

        public virtual Child? Child { get; set; } 
    }
}

File: UI/careconnect_admin/lib/models/responses/participant.dart
import 'package:careconnect_admin/models/responses/attendance_status.dart';
import 'package:careconnect_admin/models/responses/child.dart';
import 'package:careconnect_admin/models/responses/user.dart';
import 'package:careconnect_admin/models/responses/workshop.dart';
import 'package:json_annotation/json_annotation.dart';

part 'participant.g.dart'; // Don't open or edit this

@JsonSerializable()
class Participant {
  int participantId;
  DateTime registrationDate;
  DateTime modifiedDate;
  int attendanceStatusId;
  AttendanceStatus? attendanceStatus;
  User? user;
  Workshop? workshop;
  Child? child;

  Participant({
    required this.participantId,
    required this.registrationDate,
    required this.modifiedDate,
    this.attendanceStatus,
    required this.attendanceStatusId,
    this.user,
    this.workshop,
    this.child,
  });

  factory Participant.fromJson(Map<String, dynamic> json) =>
      _$ParticipantFromJson(json);

  Map<String, dynamic> toJson() => _$ParticipantToJson(this);
}
